pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationService.java

```diff
@@ -99,18 +99,38 @@ public String authenticate(AuthenticationDataSource authData, String authMethodN
     }
 
     public String authenticateHttpRequest(HttpServletRequest request) throws AuthenticationException {
-        // Try to validate with any configured provider
         AuthenticationException authenticationException = null;
         AuthenticationDataSource authData = new AuthenticationDataHttps(request);
-        for (AuthenticationProvider provider : providers.values()) {
+        String authMethodName = request.getHeader("X-Pulsar-Auth-Method-Name");
+
+        if (authMethodName != null) {
+            AuthenticationProvider providerToUse = providers
+                    .values()
+                    .parallelStream()
+                    .filter(provider -> provider.getAuthMethodName().equals(authMethodName))
+                    .findAny()
+                    .orElseThrow(() -> new AuthenticationException(String.format("Unsupported header X-Pulsar-Auth-Method-Name [%s].", authMethodName)));
```

Comment:
```suggestion
            AuthenticationProvider providerToUse = providers.get(authMethodName);
            if (providerToUse == null) {
                throw new AuthenticationException(
                    String.format("Unsupported authentication method: [%s].", authMethodName);
            }
```

Reply:
applied in 4265e40231558ce839bc429f811a4e59f2a6e8d1.